api/client/service/update.go

```diff
@@ -433,7 +444,11 @@ portLoop:
 }
 
 func equalPort(targetPort nat.Port, port swarm.PortConfig) bool {
-	return (string(port.Protocol) == targetPort.Proto() &&
+	protocol := string(port.Protocol)
+	if protocol == "" {
+		protocol = "tcp"
+	}
+	return (protocol == targetPort.Proto() &&
```

Comment:
Are you sure this is necessary?

I believe swarm always returns a spec that includes the protocol, even if one wasn't specified on the command line.